src/ImageSharp/Formats/Png/Filters/AverageFilter.cs

```diff
@@ -22,18 +22,23 @@ internal static unsafe class AverageFilter
         /// <param name="bytesPerScanline">The number of bytes per scanline</param>
         /// <param name="bytesPerPixel">The bytes per pixel.</param>
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public static void Decode(byte[] scanline, byte[] previousScanline, int bytesPerScanline, int bytesPerPixel)
+        public static void Decode(ref byte scanline, ref byte previousScanline, int bytesPerScanline, int bytesPerPixel)
```

Comment:
The reason I kept it like that was due to the encoder versions requiring the refs (We don't want to declare them twice for encoding then counting) so it felt better to use the same pattern. It does all feel hella clunky though. 

I really wish we could do some sort of clever wrapper using `Span<T>` and `#ifdefs`